docs/source/torch.compiler_troubleshooting.rst

```diff
@@ -1,724 +1,1113 @@
+3. **Non-self-contained reproducer with manageable dependencies:**
+   For example, if you can reproduce the problem by running a script after ``pip install transformers``,
+   that's manageable. We can likely run it and investigate.
+4. **Non-self-contained reproducer requiring substantial setup:**  This might involve downloading datasets,
+   multiple environment setup steps, or specific system library versions requiring a Docker image.
+   The more complex the setup, the harder it is for us to recreate the environment.
+
+   .. note::
+       Docker simplifies setup but complicates changes to the environment, so it's not a perfect solution, though we'll use it if necessary.
+
+Somewhat orthogonally, a reproducer that can be run in a single process is better than a reproducer
+that requires multiprocess training (but once again, if you only have a multiprocess reproducer, we'll take it!).
+
+Additionally, below is a non-exhaustive list of aspects to check in your
+issue that you can attempt to replicate in your reproducer:
+
+- **Autograd**. Did you have tensor inputs with ``requires_grad=True``? Did you call ``backward()`` on the output?
+- **Dynamic shapes**. Did you set ``dynamic=True``? Or did you run the test code multiple times with varying shapes?
+- **Custom operators**. Is there a custom operator involved in the real workflow?
+  Can you replicate some of its important characteristics using the Python custom operator API?
+- **Configuration**. Did you set all the same configuration?
+  This includes ``torch._dynamo.config`` and ``torch._inductor.config`` settings,
+  as well as arguments to ``torch.compile`` like ``backend`` / ``mode``.
+- **Context managers**. Did you replicate any active context managers?
+  This could be ``torch.no_grad``, automatic mixed precision, ``TorchFunctionMode`` / ``TorchDispatchMode``,
+  activation checkpointing, compiled autograd etc.
+- **Tensor subclasses**. Is there a tensor subclass involved?
+
+Minifier
+--------
+
+The minifier is an early ``torch.compile`` tool that, given an FX graph that crashes when we attempt to run or compile it,
+finds a subgraph that also crashes and outputs the code that performs that subgraph's operations.
+Essentially, the minifier finds a minimal repro for a certain class of ``torch.compile``-related crashes.
+This assumes that we were able to successfully trace through code.
+
+Unfortunately, most of the time nowadays, the minifier doesn't work as expected, and alternative methods may be necessary.
+This is likely because bugs that can be automatically reproduced in this manner are generally easier to fix
+and have already been addressed, leaving more complex issues that do not reproduce easily.
+However, it is straightforward to attempt using the minifier, so it is worth trying even if it may not succeed.
+
+Instructions for operating the minifier can be found `here <https://pytorch.org/docs/stable/torch.compiler_troubleshooting_old.html>`__.
```

Comment:
@williamwen42  This link seem to be dead. 404 in https://pytorch.org/docs/stable/torch.compiler_troubleshooting_old.html

Reply:
Try again? The link looks active to me (also on icognito).

Reply:
> Try again? The link looks active to me (also on icognito).

oh, works now! Not sure what happened when I tried it earlier